### PR TITLE
add package rocksdb (#1894)

### PR DESCRIPTION
**Pull request by byronhe**

### CODE REVIEW

File: packages/r/rocksdb/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            table.insert(configs, "DEBUG_LEVEL=2")
+        end
+
+        if package:config("shared") then
+            table.insert(configs, "shared_lib")
+        else
+            table.insert(configs, "static_lib")
+        end
+        table.insert(configs, "install")
+        import("package.tools.make").build(package, configs)
```
Comment (waruqi) [inline packages/r/rocksdb/xmake.lua:33]: it has cmakelists.txt, you should use `tools.cmake`.  tools.cmake does not support cross-compilation and switch toolchains.

File: packages/r/rocksdb/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    add_versions("v7.10.2", "4619ae7308cd3d11cdd36f0bfad3fb03a1ad399ca333f192b77b6b95b08e2f78")
+
+    add_deps("snappy")
+    add_deps("lz4")
+    add_deps("zlib")
+    add_deps("zstd")
+    add_deps("jemalloc")
+
+    if is_plat("linux") then
+        add_syslinks("pthread")
```
Comment (waruqi) [inline packages/r/rocksdb/xmake.lua:16]: add_syslinks("pthread", "rt", "dl")

File: packages/r/rocksdb/xmake.lua (hunk 1 above)
Comment (byronhe) [reply packages/r/rocksdb/xmake.lua:33]: rocksdb/CMakeLists.txt is only for Windows platform.

i tried use cmake on linux , it fails.

Comment (waruqi) [reply packages/r/rocksdb/xmake.lua:33]: if you use makefile, you need pass mingw toolchains, and dependent packages configuraion to makefile. maybe you need add much patches.

or you can use xmake.lua to port it.

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake-repo/pull/1904

**Pull request closed without merging**